src/omega_edit/server.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode'
+import * as fs from 'fs'
+import * as unzip from 'unzip-stream'
+import * as os from 'os'
+import * as child_process from 'child_process'
+import { HttpClient } from 'typed-rest-client/HttpClient'
+import XDGAppPaths from 'xdg-app-paths'
+
+const xdgAppPaths = XDGAppPaths({ name: 'omega_edit' })
+
+class Backend {
+  constructor(readonly owner: string, readonly repo: string) {}
+}
+
+class Artifact {
+  constructor(readonly omegaEditVersion: string) {}
+
+  name = `omega-edit-scala-server-${this.omegaEditVersion}`
+  archive = `${this.name}.zip`
+  archiveUrl = (backend: Backend) =>
+    `https://github.com/${backend.owner}/${backend.repo}/releases/download/v${this.omegaEditVersion}/${this.archive}`
+
+  scriptName = os.platform().toLowerCase().startsWith('win32')
+    ? 'example-grpc-server.bat'
+    : './example-grpc-server'
+
+  getOsFolder() {
+    if (os.platform().toLowerCase().startsWith('win')) {
+      return 'windows'
+    } else if (os.platform().toLowerCase().startsWith('darwin')) {
+      return 'macos'
+    } else {
+      return 'linux'
+    }
+  }
+}
+
+// Method to get omegaEditVersion from a JSON file
+export function getOmegaEditVersion(filePath: fs.PathLike) {
+  return JSON.parse(fs.readFileSync(filePath).toString())['omegaEditVersion']
+}
```

Comment:
Is this reading from the package.json file? Thoughts on making an OmegaEdit API to get the version? That way we don't need to keep this json file up to date. Whatever version of omega edit is bundled with this is what we'll get.

Reply:
Yeah we could do that or just have it read a different part of the package.json so it gets the specified version for the yarn package so they are the same. So it would read this part `"omega-edit": "0.9.7"` does that work?

Reply:
I just want to avoid having the version specified in multiple places in this repo, because we are definitely going to forget to update one of them at some point if there are multiple places to update. How that happens doesn't really matter to me. Right now it's specified in two places, once as a dependency and another as this variable.

Reply:
Yeah I will remove the variable and just read in the dependency version as they should be the same

Reply:
Fixed  this but let this as a comment but will completely remove it on my next commit that will mostly be LICENSE updates